### ISSUE TITLE
toDate works incorrectly near DST and format is not converting a date

### ISSUE THREAD

**Issue by marimano** [marnusw/date-fns-tz#44]
Hi,

I'm using `date-fns-tz` to display and parse  dates in specific timezone. 
`toDate` gives me incorrect results for period of day (3:00 - 9:59) when DST:
```
dateFnsTz.toDate(
  '2020-03-08 02:00', { timeZone: 'America/Los_Angeles' }
) // correct 2020-03-08 03:00
dateFnsTz.toDate(
'2020-03-08 03:00', { timeZone: 'America/Los_Angeles' }
) // incorrect 2020-03-08 04:00
dateFnsTz.toDate(
'2020-03-08 09:00', { timeZone: 'America/Los_Angeles' }
) // incorrect 2020-03-08 10:00
dateFnsTz.toDate(
'2020-03-08 10:00', { timeZone: 'America/Los_Angeles' }
) // correct 2020-03-08 10:00
```

Also, when I use `dateFnsTz.format`, it returns the date in local timezone rather than the specified one:
```
dateFnsTz.format(
  new Date('2020-03-03 10:00'), 
  'yyyy-MM-dd HH:mm', 
  { timeZone: 'America/Los_Angeles' }
) // incorrect "2020-03-03 10:00 -08:00", should be "2020-03-03 00:00 -08:00" as I'm in +02:00 timezone
```

**Comment by josegranja** [comment]
We are having a similar problem:

`const date = utcToZonedTime(toDate('2019-10-27T03:00:00+00:00'), 'America/New_York');
        const dateFormat = format(date, 'PPp', {
            timeZone: 'America/New_York',
        });
        expect(dateFormat).toEqual('Mar 29, 2020, 11:00 PM');`

`Expected: "Mar 29, 2020, 11:00 PM"
    Received: "Oct 27, 2019, 12:00 AM"`

something is funny around DST zones

**Comment by dioxmio** [comment]
We are having a similar problem:

`const date = utcToZonedTime(toDate('2019-10-27T03:00:00+00:00'), 'America/New_York'); const 

dateFormat = format(date, 'PPp', { timeZone: 'America/New_York', }); 

expect(dateFormat).toEqual('Oct 26, 2019, 11:00 PM');`

Expected: "Oct 26, 2019, 11:00 PM" Received: "Oct 27, 2019, 12:00 AM"

something is funny around DST

**Comment by hyyan** [comment]
@dioxmio check the latest version, this might solve your problem

**Comment by dcousens** [comment]
Alternatively,  please try https://github.com/marnusw/date-fns-tz/pull/46

**Comment by dioxmio** [comment]
@dcousens I've created this PR https://github.com/marnusw/date-fns-tz/pull/52/files where I'm showcasing the error with a possible fix. could you have a look to it? That fixes my problems

**Comment by josegranja** [comment]
@marnusw could you check this? many thanks :D

**Comment by nikolajnyboe** [comment]
Hi, 

I have run into an issue which I think is related. 

When converting a zoned date in the range **4th of April** after 4pm until the **5th of April** at 2am using the time zone "Australia/Sydney" to a UTC date, `zonedTimeToUtc` returns a date with an UTC offset of +10 hours where it should return a date with an UTC offset of +11 hours.

The time zone "Australia/Sydney" changes from **UTC+11** to **UTC+10** on the **5th of April** at 3am ([according to my knowledge](https://www.timeanddate.com/time/zone/australia/sydney)) and that could be a cause for the issue, but I don't know enough about time zone conversion to conclude that.

I have created a [CodeSandbox](https://codesandbox.io/s/strange-tree-uqzuk) to illustrate the issue.

@dcousens's PR #46 points out the same issue but with a 2019 date.

If I have used the library wrong in any way, please correct me as I only recently started using this 🙂

**Comment by lirbank** [comment]
Just ran into this issue a few days ago and for some reason assumed date-fns-tz was part of the date-fns repo (doh), so posted an issue there (https://github.com/date-fns/date-fns/issues/1691) before I found this one. date-fns seems to handle DTS correctly https://github.com/date-fns/date-fns/issues/1691#issuecomment-605716659.

Here is a few tests that show the issue:

```js
import { zonedTimeToUtc } from 'date-fns-tz';

test.each([
  ['2020-03-08 02:00', '2020-03-08T10:00:00.000Z'], // pass
  ['2020-03-08 03:00', '2020-03-08T10:00:00.000Z'], // fail
  ['2020-03-08 04:00', '2020-03-08T11:00:00.000Z'], // fail
  ['2020-03-08 05:00', '2020-03-08T12:00:00.000Z'], // fail
  ['2020-03-08 06:00', '2020-03-08T13:00:00.000Z'], // fail
  ['2020-03-08 07:00', '2020-03-08T14:00:00.000Z'], // fail
  ['2020-03-08 08:00', '2020-03-08T15:00:00.000Z'], // fail
  ['2020-03-08 09:00', '2020-03-08T16:00:00.000Z'], // fail
  ['2020-03-08 10:00', '2020-03-08T17:00:00.000Z'], // pass
])('zonedTimeToUtc(%s) -> %s', (date, expected) => {
  expect(zonedTimeToUtc(date, 'America/Los_Angeles').toISOString()).toEqual(
    expected,
  );
});
```

**Comment by marnusw** [comment]
This might be fixed in version `1.0.11` which will be available soon. Please reopen if not. If more than one issue was being discussed in this thread please open a dedicated new issue for other points if one doesn't exist already.
